data/data/agent/systemd/units/agent-register-infraenv.service.template

```diff
@@ -4,7 +4,6 @@ Wants=network-online.target assisted-service.service
 PartOf=assisted-service-pod.service
 After=network-online.target assisted-service.service agent-register-cluster.service agent-import-cluster.service
```

Comment:
Is it not like that the agent-register-infraenv.service should be the one that starts as soon as assisted service is up?

Reply:
And, agent-register-cluster.service agent-import-cluster.service should be after agent-register-infraenv.service

Reply:
For the regular ABI flow we bind the InfraEnv to the Cluster at the time the InfraEnv is created, which means the Cluster must be created first.

For the late-binding flow we can create the InfraEnv without creating a Cluster. (That's what this change enables.)

Perhaps in the future we will use late-binding everywhere, but that would require other changes to bind the hosts to the Cluster at a later stage that I am not proposing right now.